### ISSUE TITLE
Initializing dateValue Array gives type errors

### ISSUE THREAD

**Issue by tvld** [elreco/vue-tailwind-datepicker#46]
Compliments on the nice UI and clean / Tailwind design.

It's I suppose a bit work in progress. Right now we get type errors  from:

```ts
  const dateValue = ref([])
  const dateValue = ref([new Date(), new Date()])
```

What works:
```ts
const dateValue = ref({
    startDate: new Date(),
    endDate: new Date()
  })
// or:
const dateValue = ref({
    startDate: '',
    endDate: ''
  })


```

**Comment by elreco** [comment]
Thank you for your feedback! I will fix it in the next version. 😀

**Comment by tvld** [comment]
Thanks, any plans on timing? These are minor bugs I suppose... ?

**Comment by tvld** [comment]
This is work-around by the way:

```ts

<template>
  <div>
    <vue-tailwind-datepicker v-model="dateValue" i18n="nl" no-input as-single :auto-apply="false" />
    {{ parsedDate }}
  </div>
</template>

<script setup lang="ts">
  import { ref } from 'vue'
  import { parse } from 'date-fns'
  import VueTailwindDatepicker from 'vue-tailwind-datepicker'

  const dateValue = ref('')

  const parsedDate = computed(() => {
    return parse(dateValue.value, 'yyyy-MM-dd kk:mm:ss', new Date())
  })
</script>
```

**Comment by elreco** [comment]
@tvld what version do you use please? I can't reproduce your type error

My tsconfig.json:

```
{
  "compilerOptions": {
    "module": "ESNext",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "useDefineForClassFields": true,

    // Required in Vue projects
    "jsx": "preserve",

    // `"noImplicitThis": true` is part of `strict`
    // Added again here in case some users decide to disable `strict`.
    // This enables stricter inference for data properties on `this`.
    "noImplicitThis": true,
    "strict": true,

    // Required in Vite
    "isolatedModules": true,
    // For `<script setup>`
    // See <https://devblogs.microsoft.com/typescript/announcing-typescript-4-5-beta/#preserve-value-imports>
    "preserveValueImports": true,
    // Enforce using `import type` instead of `import` for types
    "importsNotUsedAsValues": "error",

    // A few notes:
    // - Vue 3 supports ES2016+
    // - For Vite, the actual compilation target is determined by the
    //   `build.target` option in the Vite config.
    //   So don't change the `target` field here. It has to be
    //   at least `ES2020` for dynamic `import()`s and `import.meta` to work correctly.
    // - If you are not using Vite, feel free to override the `target` field.
    "target": "ESNext",

    // Recommended
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    // See <https://github.com/vuejs/vue-cli/pull/5688>
    "skipLibCheck": true,
  }
}
```

**Comment by tvld** [comment]
I doubt it's the tsconfig, but this is mine:
```json
"compilerOptions": {
    "allowJs": true,
    "strict": true,
    "noImplicitThis": true
  },
```

**Comment by tvld** [comment]
Version in package.json   "vue-tailwind-datepicker": "^1.2.9",

**Comment by tvld** [comment]
You forgot to include Date:
```log
(property) modelValue: string | [Date, Date] | {
    start: string | Date;
    end: string | Date;
} | {
    startDate: string | Date;
    endDate: string | Date;
}
Type 'Date' is not assignable to type 'string | [Date, Date] | { start: string | Date; end: string | Date; } | { startDate: string | Date; endDate: string | Date; }'.
  Type 'Date' is not assignable to type '[Date, Date]'.ts(2322)
types.d.ts(42, 5): The expected type comes from property 'modelValue' which is declared here on type 'ComponentProps<VueTailwindDatePicker>'
```

**Comment by xibman** [comment]
@tvld should be fixed with 1.5.1
